web-gui/src/components/GeneratePGP.jsx

```diff
@@ -0,0 +1,154 @@
+import React, { useState, forwardRef, Suspense, useMemo } from 'react';
+import Stack from '@mui/material/Stack';
+import Button from '@mui/material/Button';
+import { generateKey, readKey } from 'openpgp/lightweight';
+
+import Grid from '@mui/material/Grid';
+import PublicKey from './PublicKey'
+import TextField from '@mui/material/TextField';
+import { saveAs } from 'file-saver';
+import JSzip from 'jszip'
+import { useContractWrite, usePrepareContractWrite } from 'wagmi'
+import KVStore from '../contracts/KVStore.json'
+import { NFTStorage } from 'nft.storage'
+import Snackbar from '@mui/material/Snackbar';
+import MuiAlert from '@mui/material/Alert';
+
+import { wrap } from 'comlink';
+import MyComlinkWorker,{api} from './generate.worker';
+const Alert = forwardRef(function Alert(props, ref) {
+  return <MuiAlert elevation={6} ref={ref} variant="filled" {...props} />;
+});
+
+const client = new NFTStorage({ token: process.env.REACT_APP_NFT_STORAGE_API })
+export default function BasicButtons() {
+  const [open, setOpen] = useState(false);
+
+
+
+  const handleClose = (event, reason) => {
+    if (reason === 'clickaway') {
+      return;
+    }
+
+    setOpen(false);
+  };
+  const [name, setName] = useState()
+  const [email, setEmail] = useState()
+  const [password, setPassword] = useState()
+  const [privateKey, setPrivateKey] = useState()
+  const [publicKey, setPublicKey] = useState()
+  const [fingerprint, setFingerprint] = useState()
+  const [loading, setLoading] = useState(false)
+  const [cid, setCid] = useState(false)
+
+
+  const { config } = usePrepareContractWrite({
+    addressOrName: process.env.REACT_APP_CONTRACT,
+    contractInterface: KVStore.abi,
+    functionName: 'setPGP',
+    args: [fingerprint, cid]
+  })
+  const { write } = useContractWrite({
+    ...config, onSuccess(data) {
+      setOpen(true)
+    }
+  })
+  async function generatePublicKey() {
+    
+setPublicKey(false)
+    setLoading(true)
+    try {
+      // const myComlinkWorkerInstance = new MyComlinkWorker({type:"module",name:new Date().getTime()});
+      // const myComlinkWorkerApi = wrap(myComlinkWorkerInstance);
+      // const data=await myComlinkWorkerApi.generate(name,email,password);
+      // const { privateKey: privkey, publicKey: pubkey }=data;
+      // myComlinkWorkerInstance.terminate()
+      // const { privateKey: privkey, publicKey: pubkey } = await generateKey({
+      //   type: 'rsa', // Type of the key
+      //   rsaBits: 2048, // RSA key size (defaults to 4096 bits)
+      //   userIDs: [{ name, email }],
+      //   passphrase: password
+
+      // })
+      const res = await fetch('/api/generate', {
```

Comment:
It's preferable to create a service for all API calls and extract this from the logic of the component